### PR TITLE
Create API for Previous Employments with Create, View and Update methods

### PR DESCRIPTION
**Pull request by mohinid**
## Notion card
https://www.notion.so/saeloun/Create-API-for-Employment-Details-under-Profile-67d88c25843c4e58a4b02edcd57a6fc5

## Summary
Create API for Previous Employments with Create, View, and Update methods for the Employee Details page under profile.
Test on Postman
Create RSPecs.

## Preview
https://www.notion.so/saeloun/Create-API-for-Employment-Details-under-Profile-67d88c25843c4e58a4b02edcd57a6fc5#7ad431e290574fed9a8925c529edb446

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?
With Postman and RSpec.

### Checklist:

- [x] I have manually tested all workflows
- [x] I have performed a self-review of my own code
- [x] I have added automated tests for my code

### CODE REVIEW

**Comment by mohinid** [comment]
Verified Postman response for GET and PATCH requests.

Screenshot 2022-07-13 at 1 02 58 PM
 requests.
Screenshot 2022-07-13 at 1 02 49 PM

**Comment by supriya3105** [comment]
@mohinid Please resolve conflicts

**Comment by mohinid** [comment]
> @mohinid Please resolve conflicts

done.

File: app/controllers/internal_api/v1/team_members/previous_employments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  def update
+    authorize employment, policy_class: TeamMembers::PreviousEmploymentPolicy
+    user = employment.user
+    previous_employments = user.previous_employments
+    previous_employments.update!(previous_employment_params)
+    render json: {
+      previous_employments:,
+      notice: ("Previous employment updated successfully.")
+    }, status: :ok
```
Comment (harshaanNihal) [inline app/controllers/internal_api/v1/team_members/previous_employments_controller.rb:17]: Either render through `update.json.jbuilder` by passing locals or remove this file[app/views/internal_api/v1/team_members/previous_employments/update.json.jbuilder]

File: app/views/internal_api/v1/team_members/previous_employments/update.json.jbuilder [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# frozen_string_literal: true
+
+json.partial! "previous_employment", locals: { user: }
```
Comment (harshaanNihal) [inline app/views/internal_api/v1/team_members/previous_employments/update.json.jbuilder:3]: https://github.com/saeloun/miru-web/pull/564/files#r923134862

**Review by harshaanNihal — APPROVED** [review]
Added a comment rest looks good to me. ✨

File: app/controllers/internal_api/v1/team_members/previous_employments_controller.rb (hunk 1 above)
Comment (harshaanNihal) [reply app/controllers/internal_api/v1/team_members/previous_employments_controller.rb:17]: Prefer using update.json.jbuilder file for consistency as we are using the same for show method.

**Comment by supriya3105** [comment]
@akhilgkrishnan Can you review this PR.

File: app/controllers/internal_api/v1/previous_employments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  def update
+    authorize previous_employment
+    previous_employment.update!(previous_employment_params)
+    render :update, locals: { previous_employment: }, status: :ok
+  end
+
+  def create
+    authorize PreviousEmployment
+    previous_employment = PreviousEmployment.create!(previous_employment_params.merge(user_id: current_user.id))
```
Comment (keshavbiswa) [inline app/controllers/internal_api/v1/previous_employments_controller.rb:17]:
```suggestion
    authorize PreviousEmployment
    previous_employment = current_user.previous_employments.create!(previous_employment_params)
```

**Review by keshavbiswa — APPROVED** [review]

**Pull request merged**